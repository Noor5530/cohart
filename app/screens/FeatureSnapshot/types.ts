import { UserState } from "models/reducers/user";
export type Post = {
  _id: string | number;
  user_id: string | number;
  created_at: string;
  updated_at: string;
  title: string;
  description: string;
  tags: string;
  processing_status: string;
  post_image: string;
  post_file: string;
  jw_media_id: string;
  jw_media_url: string;
  jw_media_thumb: string;
  post_type: string;
  is_hide: boolean;
  created_by: UserState;
  glow_count: number;
  is_featured: boolean;
};

type user = {};

export interface FeatureSnapshotProps {
  item: Post;
}
