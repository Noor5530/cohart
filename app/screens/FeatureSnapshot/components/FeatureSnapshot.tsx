import React from "react";
import { useNavigation } from "@react-navigation/native";
import { FeatureSnapshotProps } from "../types";
import { widthPercentageToDP } from "react-native-responsive-screen";
import AnimatedImage from "components/AnimatedImage";
import { TouchableOpacity, StyleSheet } from "react-native";
import { shimmerColor } from "../../../config/colors";
const FeatureSnapshotList = (props: FeatureSnapshotProps) => {
  const { item } = props;
  const navigation = useNavigation();
  const onPress = () => {
    navigation.navigate("NewPostPreview", {
      post: item,
      back: true,
      from: "featureSnapshot",
    });
  };
  return (
    <TouchableOpacity
      testID="featureSnapShots"
      onPress={onPress}
      style={styles.container}
    >
      <AnimatedImage
        thumbnail={item?.jw_media_thumb}
        style={styles.image}
        resizeMode="cover"
        uri={item?.jw_media_thumb}
      />
    </TouchableOpacity>
  );
};

export default FeatureSnapshotList;

const styles = StyleSheet.create({
  container: {
    height: 160,
    width: widthPercentageToDP(31),
    backgroundColor: "white",
  },
  image: { flex: 1, margin: 5, backgroundColor: shimmerColor },
});
