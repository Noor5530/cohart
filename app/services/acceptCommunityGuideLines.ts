import ApiConfig from "config/api-config";
import { apiClient } from "services/client";

export const acceptCommunityGuideLines = (data: any) => {
  return apiClient.post(ApiConfig.UPDATE_USER_GUIDE, data);
}
