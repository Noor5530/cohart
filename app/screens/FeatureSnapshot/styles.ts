import { appFonts } from "components/text";
import { StyleSheet } from "react-native";
const styles = StyleSheet.create({
  container: { backgroundColor: "white", flex: 1 },
  listContainer: {
    alignItems: "center",
    backgroundColor: "white",
  },
  listComponent: {
    alignSelf: "center",
    alignItems: "center",
    justifyContent: "center",
    paddingTop: 20,
  },
  emptyText: {
    fontSize: 14,
    fontFamily: appFonts.InterBlack,
  },
});
export default styles;
