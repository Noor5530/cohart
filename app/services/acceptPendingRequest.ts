import ApiConfig from 'config/api-config';
import { apiClient } from 'services/client';

export const acceptPendingRequest = (data: object) => {
  return apiClient.post(ApiConfig.ACCEPT_PENDING_REQUEST, data);
}
