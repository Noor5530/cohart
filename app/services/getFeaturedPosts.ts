import ApiConfig from "config/api-config";
import { apiClient } from "services/client";

type GetFeaturePostsPrams = {
  admin_view: boolean;
};
export function getFeaturedPosts(data: GetFeaturePostsPrams) {
  return apiClient.get(ApiConfig.GET_FEATURED_POSTS, {
    params: data,
  });
}
