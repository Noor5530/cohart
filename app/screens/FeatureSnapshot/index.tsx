import React, { useEffect, useState, useCallback } from "react";

import { useDispatch } from "react-redux";
import { getFeaturedPosts } from "services/getFeaturedPosts";
import { enableSnackBar } from "../../store/actions/snackBarAction";
import { useIsFocused } from "@react-navigation/native";
import FeatureSnapshot from "./components/FeatureSnapshot";
import { FlatList, View, Text } from "react-native";
import Header from "components/Header";
import styles from "./styles";
import { Post } from "./types";
const FeatureSnapshotList = () => {
  const [featureSnapShort, setFeatureSnapShort] = useState([]);
  const [loading, setLoading] = useState(false);
  const isFocused = useIsFocused();
  const dispatch = useDispatch();
  useEffect(() => {
    if (isFocused) {
      fetchFeatureSnapShort();
    }
  }, [isFocused]);
  const fetchFeatureSnapShort = useCallback(async () => {
    try {
      setLoading(true);
      let response = await getFeaturedPosts({ admin_view: true });
      setFeatureSnapShort(response?.data?.data);
      setLoading(false);
    } catch (error) {
      console.log(error);
      dispatch(enableSnackBar());
      setLoading(false);
    }
  }, [setLoading, setFeatureSnapShort]);
  const renderItem = ({ item }: { item: Post }) => {
    return <FeatureSnapshot item={item} />;
  };
  const ListEmptyComponent = () => {
    return (
      <View style={styles.listComponent}>
        {loading ? null : <Text style={styles.emptyText}>No post found</Text>}
      </View>
    );
  };
  return (
    <View style={styles.container}>
      <Header back />
      <FlatList
        testID={"SnapShotList"}
        onRefresh={fetchFeatureSnapShort}
        refreshing={loading}
        style={styles.container}
        contentContainerStyle={styles.listContainer}
        ListEmptyComponent={ListEmptyComponent}
        keyExtractor={(item, index) => `id_${index}`}
        data={featureSnapShort}
        numColumns={3}
        renderItem={renderItem}
      />
    </View>
  );
};

export default FeatureSnapshotList;
