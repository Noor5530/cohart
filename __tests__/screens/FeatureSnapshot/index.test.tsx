import "react-native";
import React from "react";
import FeatureSnapshot from "../../../app/screens/FeatureSnapshot";
import renderer from "react-test-renderer";
import { render, fireEvent, act } from "@testing-library/react-native";
import * as featuredPosts from "services/getFeaturedPosts";
import { AxiosResponse } from "axios";
import * as reactNavigation from "@react-navigation/native";
jest.useFakeTimers();

test("renders correctly", () => {
  const tree = renderer.create(<FeatureSnapshot />).toJSON();
  act(() => {
    jest.spyOn(reactNavigation, "useIsFocused").mockImplementation(() => true);
  });
  expect(tree).toMatchSnapshot();
});

test("get featureSnapshot ", () => {
  const { getByTestId } = render(<FeatureSnapshot />);

  act(() => {
    const mockResponse: AxiosResponse = {
      data: {
        data: [
          {
            _id: "1111111111111",
          },
        ],
        statusCode: 200,
      },
      status: 200,
      statusText: "",
      headers: undefined,
      config: {},
    };

    const mockGetFeaturedPosts = jest
      .spyOn(featuredPosts, "getFeaturedPosts")
      .mockImplementation(() => Promise.resolve(mockResponse));
    fireEvent(getByTestId("SnapShotList"), "onRefresh");
    expect(mockGetFeaturedPosts).toHaveBeenCalled();
  });
});
test("fail on get  featureSnapshot", () => {
  const { getByTestId } = render(<FeatureSnapshot />);

  act(() => {
    const mockGetFeaturedPosts = jest
      .spyOn(featuredPosts, "getFeaturedPosts")
      .mockImplementation(() => Promise.reject());
    act(() => {
      fireEvent(getByTestId("SnapShotList"), "onRefresh");
    });
    expect(mockGetFeaturedPosts).toHaveBeenCalled();
  });
});
