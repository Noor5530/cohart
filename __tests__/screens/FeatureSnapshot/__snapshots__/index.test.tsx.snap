// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingTop": 30,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "width": 38,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="14"
          bbWidth="21"
          fill="none"
          focusable={false}
          height="14"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 14,
                "width": 21,
              },
            ]
          }
          vbHeight={14}
          vbWidth={21}
          width="21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M6.64474 0.583984L0.433134 6.79559L0.167969 7.06076L0.433134 7.32592L6.64474 13.5375L7.17507 13.0072L1.66787 7.5L21 7.5V6.75L1.53939 6.75L7.17507 1.11431L6.64474 0.583984Z"
              fill={4278190080}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 20,
            },
            Object {},
            Object {
              "fontFamily": "AktivGroteskExTrial-Bold",
            },
          ]
        }
      >
        COHART
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="19"
        bbWidth="16"
        fill="none"
        focusable={false}
        height="19"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 19,
              "width": 16,
            },
          ]
        }
        vbHeight={19}
        vbWidth={16}
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M7.58726 1.75C8.01448 1.75 8.36081 2.09633 8.36081 2.52355C8.36081 3.03874 8.66732 3.59741 9.26471 3.7987C11.462 4.53906 13.0903 6.71461 13.0903 9.32316C13.0903 9.36104 13.0899 9.39882 13.0893 9.43649C13.0881 9.50183 13.0931 9.56582 13.1039 9.62789V12.5105C13.1039 12.8394 13.1805 13.1637 13.3275 13.4579L14.1596 15.1219C14.199 15.2007 14.1417 15.2934 14.0536 15.2934H1.11992C1.03185 15.2934 0.974564 15.2007 1.01395 15.1219L1.81191 13.526C1.95899 13.2319 2.03556 12.9075 2.03556 12.5786V9.75294C2.06936 9.64853 2.08682 9.53692 2.08501 9.42124C2.0845 9.38862 2.08424 9.35593 2.08424 9.32316C2.08424 6.71462 3.71252 4.53907 5.90981 3.7987C6.5072 3.59741 6.81371 3.03874 6.81371 2.52355C6.81371 2.09633 7.16004 1.75 7.58726 1.75ZM9.58401 2.85105C9.44664 2.80476 9.36081 2.66851 9.36081 2.52355C9.36081 1.54405 8.56676 0.75 7.58726 0.75C6.60775 0.75 5.81371 1.54405 5.81371 2.52355C5.81371 2.66851 5.72787 2.80476 5.5905 2.85105C2.97586 3.73204 1.08424 6.29665 1.08424 9.32316C1.08424 9.36116 1.08454 9.39908 1.08513 9.43693C1.08535 9.45079 1.07419 9.46217 1.06033 9.46217C1.04665 9.46217 1.03556 9.47326 1.03556 9.48694V12.5786C1.03556 12.7523 0.995133 12.9235 0.91748 13.0788L0.119523 14.6747C-0.252313 15.4184 0.288463 16.2934 1.11992 16.2934H5.15932C5.40744 17.4128 6.406 18.25 7.6001 18.25C8.79419 18.25 9.79276 17.4128 10.0409 16.2934H14.0536C14.8851 16.2934 15.4258 15.4184 15.054 14.6747L14.222 13.0107C14.1443 12.8554 14.1039 12.6841 14.1039 12.5105V9.46957C14.1039 9.46549 14.1006 9.46217 14.0965 9.46217C14.0923 9.46217 14.089 9.45878 14.0891 9.45464C14.0899 9.41091 14.0903 9.36709 14.0903 9.32316C14.0903 6.29665 12.1987 3.73203 9.58401 2.85105ZM8.99864 16.2934H6.20155C6.41923 16.8532 6.96332 17.25 7.6001 17.25C8.23687 17.25 8.78097 16.8532 8.99864 16.2934Z"
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
      }
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
    testID="SnapShotList"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Inter-Black",
              "fontSize": 14,
            }
          }
        >
          No post found
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
